.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -3133,12 +3130,6 @@ stages:
         artifact: benchmarks_results_BenchmarkAgent6
         path: $(System.DefaultWorkingDirectory)/tracer/build_data/benchmarks
 
-    - task: DownloadPipelineArtifact@2
-      displayName: Download results BenchmarkAgent7
-      inputs:
-        artifact: benchmarks_results_BenchmarkAgent7
-        path: $(System.DefaultWorkingDirectory)/tracer/build_data/benchmarks
-
```

Comment:
This is interesting 🤔 We run the comparison _twice_, once for the asm stage, once for the tracer stage. However, because the comparison _replaces_ the existing comment on the PR, I think we need to make some more changes 🤔 

Otherwise, whichever stage finishes first (let's say tracer) will update the PR without the ASM results. Then the ASM stage will run and replace the PR comment with _only_ the ASM results.

So _either_ we need to:
- _Always_ download _all_ of the results, in _both_ stages. The first time, not all the results are there, the second time they all will be. You need to "allow failure" when downloading the artifact from the other stage in that case.
- Re-write the  benchmark comparison code to write the results as two separate comments. This might be the easiest/least confusion option - pass the `BenchmarkCategory` as an arg when calling `CompareBenchmarksResults`, and use it in that target to change the markdown title (and hence change the "replacement" behaviour). Happy to describe this in more detail if needs be 😄 

FYI, one thing you _can't_ do is move the comparison to a separate stage that depends on these two, because that stage would never run unless _both_ benchmark stages run, which would defeat the purpose of this PR😄